utils/funcexp/func_from_unixtime.cpp

```diff
@@ -67,7 +67,7 @@ DateTime getDateTime(rowgroup::Row& row,
 
             if (parm[0]->data()->resultType().colWidth == datatypes::MAXDECIMALWIDTH)
             {
-                auto integralAndFractional = dec.getIntegralAndFractional<datatypes::TSInt128>();
+                auto integralAndFractional = dec.getIntegralAndFractional();
```

Comment:
I think this code is wrong. Both the narrow decimal and the wide decimal branch:
```
DROP TABLE IF EXISTS t1;
CREATE TABLE t1 (a DECIMAL(10,7)) ENGINE=ColumnStore;
INSERT INTO t1 VALUES (1.3);
SELECT FROM_UNIXTIME(a) FROM t1;
+----------------------------+
| FROM_UNIXTIME(a)           |
+----------------------------+
| 1970-01-01 04:00:01.902848 |
+----------------------------+
```

The correct result is:
```
DROP TABLE IF EXISTS t1;
CREATE TABLE t1 (a DECIMAL(10,7)) ENGINE=MyISAM;
INSERT INTO t1 VALUES (1.3);
SELECT FROM_UNIXTIME(a) FROM t1;
+----------------------------+
| FROM_UNIXTIME(a)           |
+----------------------------+
| 1970-01-01 04:00:01.300000 |
+----------------------------+
```

The problem is that the code does not take into account how the scale relates to 6 (which is the divisor  for microsecords).
The code is correct when scale is exactly 6.
But it's wrong for other scale values.

Reply:
Agree but this goes out of the scope.

Reply:
I'll create an issue for this.